docs: Add development installation instructions

KennethEnevoldsen opened the pull request:
Added missing development installation instructions.

imenelydiaker commented on README.md:
````diff
@@ -213,6 +214,30 @@ Advanced scripts with different models are available in the [mteb/mtebscripts re
+```bash
+git clone https://github.com/embeddings-benchmark/mteb
+pip install -e .
+pip install -r requirement.dev.txt
````
along with `pip install -re requirements.txt` because it contains the packages needed to run mteb no?

KennethEnevoldsen replied:
as least when I tested it it was not required.

imenelydiaker commented on README.md:
````diff
@@ -213,6 +214,30 @@ Advanced scripts with different models are available in the [mteb/mtebscripts re
+```bash
+git clone https://github.com/embeddings-benchmark/mteb
+pip install -e .
+pip install -r requirement.dev.txt
````
```suggestion
pip install -r requirements.dev.txt
```

imenelydiaker commented on README.md:
````diff
@@ -213,6 +214,30 @@ Advanced scripts with different models are available in the [mteb/mtebscripts re
+### Running Tests
+To run the tests, you can use the following command:
+
+```bash
+pytest 
+# or if you want to run on multiple cores
+pytest -n auto
+```
````
There is a make file here: https://github.com/embeddings-benchmark/mteb/blob/main/Makefile
```suggestion
```bash
make test
```
```

KennethEnevoldsen replied:
Then we should rewrite make test (it currently takes up all cores which at least for me led to it hanging).

KennethEnevoldsen commented:
I have now addtionally:

1) updated the CI to use the make commands (this allows the user a 1-1 mapping between ci and make commands)
2) Added linting to docs

imenelydiaker commented on .github/workflows/python-package.yml:
```diff
@@ -27,15 +27,11 @@ jobs:
-        python -m pip install flake8 pytest
         python -m pip install .
-        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+        python -m pip install -r requirements.dev.txt
```
I think that you should keep the requirements.txt, else the current package won't find the needed dependencies to run. Why do you think it's not useful ?

KennethEnevoldsen replied:
the requirements are already specified in the setup.py which is installed as a part of `python -m pip install .`

Muennighoff replied:
We could also remove the `requirements.dev.txt` and have them as an extra similar to [transformers](https://github.com/huggingface/transformers/blob/00c1d87a7d5c8dfb4554370983b5a3f7c069edd7/setup.py#L445C20-L445C26) ; So one would install dev with something like `pip install -e ".[dev]"` (https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md); I don't mind either one, the current way may actually use less code & is simpler so I think it's fine

KennethEnevoldsen replied:
I will change it to the ".[dev]" approach. That was all requirements are specified in one place.

Muennighoff commented on Makefile:
```diff
@@ -13,25 +13,38 @@ modified_only_fixup:
\ No newline at end of file
+	pytest
+
+# add parllel test for faster execution (can sometimes cause issues with some tests)
+test-parallel:
+	pytest -n auto --dist=loadfile -s -v
```
```suggestion
# add parallel test for faster execution (can sometimes cause issues with some tests)
test-parallel:
	pytest -n auto --dist=loadfile -s -v

```

Muennighoff commented on README.md:
```diff
@@ -213,6 +214,39 @@ Advanced scripts with different models are available in the [mteb/mtebscripts re
+We welcome contributions such as new datasets to MTEB! This section describes how to set up the repository for development.
+
+### Development Installation
+If you want to submit a dataset or on other ways contribute to MTEB, you can install the package in development mode:
```
```suggestion
If you want to submit a dataset or contribute in other ways to MTEB, you can install the package in development mode:
```

Muennighoff reviewed (approved):
Really nice work; should make contributing much easier!

Muennighoff commented:
Are the failing tests expected?

KennethEnevoldsen commented:
> Are the failing tests expected?

No. Not sure why it happens. Will try to debug it
